src/main/java/org/tridiots/ipc/SocketObjectUtil.java

```diff
@@ -0,0 +1,55 @@
+package org.tridiots.ipc;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SocketObjectUtil {
+    private static final Logger logger = LoggerFactory.getLogger(SocketObjectUtil.class);
+    private static final int DATA_LENGTH = 1024;
+
+    public static Object receiveObject(SocketChannel channel) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        ByteBuffer buffer = ByteBuffer.allocate(DATA_LENGTH);
+
+        byte[] bytes = null;
+        int size = 0;
+        try {
+            while ((size = channel.read(buffer)) > 0) {
+                bytes = new byte[size];
+                buffer.flip();
+                buffer.get(bytes);
+                baos.write(bytes);
```

Comment:
`buffer.array()` represent the whole array, and there is function as you said, then we can modify it to`baos.write(buffer.array(), 0, size);`